connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala

```diff
@@ -1567,6 +1613,19 @@ class SparkConnectPlanner(val session: SparkSession) {
         val expr = transformExpression(fun.getArguments(0))
         Some(transformUnregisteredUDF(MLFunctions.arrayToVectorUdf, Seq(expr)))
 
+      // Protobuf-specific functions
+      case "from_protobuf" if Seq(2, 3, 4).contains(fun.getArgumentsCount) =>
+        val children = fun.getArgumentsList.asScala.toSeq.map(transformExpression)
+        val (messageClassName, descFilePathOpt, options) =
+          extractArgsOfProtobufFunction("from_protobuf", fun.getArgumentsCount, children)
+        Some(ProtobufDataToCatalyst(children.head, messageClassName, descFilePathOpt, options))
```

Comment:
Is it better to call `from_protobuf()` rather than invoking `ProtobufDataToCatalyst` directly? 
Any future changes for `from_protobuf()` will apply automatically to these.

Reply:
In my impression, @hvanhovell  mentioned that the code in  `transformUnregisteredFunction` will be uniformly processed to reuse `FunctionRegistry` in the future. Maybe we can wait for a unified solution?

Reply:
Thanks @LuciferYang. This sounds good. It goes with the rest of the implementations too. We will update here when we update legacy functions.